File: java/rotational-cipher/src/main/java/RotationalCipher.java
class RotationalCipher {

    private final int shiftKey;

    private final static int NUMBER_OF_LETTERS = 26;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        StringBuilder str = new StringBuilder();

        for(Character ch : data.toCharArray()){
            str.append(rotateSymbol(ch));
        }

        return str.toString();
    }

    private Character rotateSymbol(char symbol){
        if(symbol >= 'a' && symbol <= 'z'){
            symbol += shiftKey;
            if(symbol > 'z')
                symbol -= NUMBER_OF_LETTERS;
            return  symbol;
        } else if(symbol >= 'A' && symbol <= 'Z'){
            symbol += shiftKey;
            if(symbol > 'Z')
                symbol -= NUMBER_OF_LETTERS;
            return  symbol;
        }

        return symbol;
    }

}

File: csharp/rotational-cipher/RotationalCipher.cs
using System.Linq;

public static class RotationalCipher
{
    private const int AlphabetSize = 26;

    public static string Rotate(string text, int shiftKey)
         => new string(text.Select(c => RotateChar(c, shiftKey)).ToArray());

    private static char RotateChar(char c, int key)
    {
        if (c >= 'a' && c <= 'z')
        {
            return (char)('a' + RotateInt(c - 'a', key));
        }

        if (c >= 'A' && c <= 'Z')
        {
            return (char)('A' + RotateInt(c - 'A', key));
        }

        return c;
    }

    private static int RotateInt(int c, int key) => (c + key) % AlphabetSize;
}